schema/documents.go

```diff
@@ -5,4 +5,5 @@ type Document struct {
 	PageContent string
 	Metadata    map[string]any
 	Score       float32
+	CustomID    *string
```

Comment:
I'm thinking we should not use a pointer here and treat the empty string as not provided.